crates/pyo3/src/analysis.rs

```diff
@@ -195,19 +221,33 @@ impl PyEventLog {
         m.into_iter().collect()
     }
 
+    fn begin(&mut self, start: Option<i64>) {
+        self.start = start;
```

Comment:
@dorschs57 I was debating having these setters (`begin`, `until`) mutate state vs return a copy.  Does one way work better with the way the UI is tracking the loaded event log?

Reply:
@jw3 I refreshed my memory on the analysis page and I think we're good with mutating it here. The use case I can see for this would be if we ever changed our implementation so you could flip back and forth between the analysis view and something else, and you wanted the temporal filter to be persisted. Mutating the globally stored state would allow for that.

Reply:
👍